src/aws_lc_rs_algs.rs

```diff
@@ -39,6 +45,21 @@ impl SignatureVerificationAlgorithm for AwsLcRsAlgorithm {
     }
 }
 
+/// Restricts the allowed encodings of EC public keys.
+///
+/// "The first octet of the OCTET STRING indicates whether the key is
+///  compressed or uncompressed.  The uncompressed form is indicated
+///  by 0x04 and the compressed form is indicated by either 0x02 or
+///  0x03 (see 2.3.3 in \[SEC1\]).  The public key MUST be rejected if
+///  any other value is included in the first octet."
+/// -- <https://datatracker.ietf.org/doc/html/rfc5480#section-2.2>
+fn ec_compressed_or_uncompressed(public_key: &[u8]) -> Result<(), InvalidSignature> {
+    match public_key.first() {
+        Some(&0x04) | Some(&0x02) | Some(&0x03) => Ok(()),
```

Comment:
I would expect you might be able to do without the `&` prefixes in the patterns here?

Reply:
AHHH, when did that start working?

Reply:
oh good grief 🤕